ouroboros-consensus/src/Ouroboros/Consensus/NodeKernel.hs

```diff
@@ -146,7 +145,7 @@ initNodeKernel args@NodeArgs { registry, cfg, tracers, maxTxCapacityOverride
 
     st <- initInternalState args
 
-    whenJust blockForging $ forkBlockForging maxTxCapacityOverride st
+    mapM_ (forkBlockForging maxTxCapacityOverride st) blockForging
```

Comment:
The main change: we spawn a separate thread for each `BlockForging`. This means each thread will do the same checks, including ticking the chain, which can be expensive.